addons/stock/models/stock_move_line.py

```diff
@@ -658,18 +658,35 @@ def _get_aggregated_product_quantities(self, **kwargs):
             uom = move_line.product_uom_id
             line_key = str(move_line.product_id.id) + "_" + name + (description or "") + "uom " + str(uom.id)
 
+            qty_ordered = move_line.move_id.product_uom_qty
+            # Loop to get backorders, and backorders' backorders, and so and so...
+            backorders = self.env['stock.picking']
+            current_picking = move_line.picking_id
+            while current_picking.backorder_ids:
+                backorders |= current_picking.backorder_ids
+                current_picking = current_picking.backorder_ids
+            if backorders:
+                # Adds the quantities delayed to backorders in order to retrieve the original ordered qty.
+                following_move_lines = backorders.move_lines.filtered(lambda ml: ml.product_id == move_line.product_id)
+                qty_ordered += sum(following_move_lines.mapped('product_uom_qty'))
```

Comment:
in case of multiple stock move lines for the same stock move, you will add multiple times the `product_uom_qty`